akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -38,7 +40,14 @@ final class Flow[-In, +Out, +Mat](private[stream] override val module: Module)
 
   private[stream] def isIdentity: Boolean = this.module eq GraphStages.Identity.module
 
-  override def via[T, Mat2](flow: Graph[FlowShape[Out, T], Mat2]): Repr[T] = viaMat(flow)(Keep.left)
+  override def via[T, Mat2](flow: Graph[FlowShape[Out, T], Mat2]): Repr[T] =
+    if (this.isIdentity) {
+      import Predef.Map.empty
+      import StreamLayout.{ CompositeModule, Ignore, IgnorableMatValComp, Transform, Atomic }
+      val m = flow.module
+      val mat = if (IgnorableMatValComp(m)) Ignore else Transform(_ => NotUsed, Atomic(m))
+      new Flow(CompositeModule(Set(m), m.shape, empty, empty, mat, Attributes.none))
```

Comment:
This is another optimization: if we always delegate to `viaMat` this will always result in a computation node to be added, which will prevent the computation from being ignored, even if it could be.